src/hotspot/os/linux/os_linux.cpp

```diff
@@ -3564,6 +3564,28 @@ bool os::Linux::hugetlbfs_sanity_check(bool warn, size_t page_size) {
   return result;
 }
 
+bool os::Linux::shm_hugetlbfs_sanity_check(bool warn, size_t page_size) {
+  // Try to create a large shared memory segment.
+  int shmid = shmget(IPC_PRIVATE, page_size, SHM_HUGETLB|IPC_CREAT|SHM_R|SHM_W);
+  if (shmid == -1) {
+    // Possible reasons for shmget failure:
+    // 1. shmmax is too small for the request.
+    //    > check shmmax value: cat /proc/sys/kernel/shmmax
+    //    > increase shmmax value: echo "0xffffffff" > /proc/sys/kernel/shmmax
```

Comment:
I'd avoid using a 32 bit constant on potentially 64 bit systems. According to that same man page, the limit on 64 bits is 127 TB which is more than the suggested 4 GB :)

The man page talks about `ULONG_MAX - 2^24` as default (for 64 bit systems?).

Reply:
The comment is a copy of the old one where we actually use `shmget()`, but I agree that it could use an update. Not even sure we have to suggest a specific value, something like:
```suggestion
    //    > increase shmmax value: echo "new value" > /proc/sys/kernel/shmmax
```
Might be good enough. 

I will also update the original comment.